Form/Type/Select2EntityType.php

```diff
@@ -107,6 +108,37 @@ public function finishView(FormView $view, FormInterface $form, array $options)
             }
         }
 
+        if ($view->vars['allow_add']['enabled'] && $form->isSubmitted()) {
+            // Form is being displayed again after a submit that failed validation.
+            // $view->vars['value'] needs to be rebuilt to handle new entities added with a tag.
+
+            // get entities as iterable collection or array.
+            $entities = $options['multiple'] ? $form->getData() : [$form->getData()];
+
+            $accessor = PropertyAccess::createPropertyAccessor();
+            $textProperty = isset($options['text_property']) ? $options['text_property'] : null;
+            $newTagPrefix = $view->vars['allow_add']['new_tag_prefix'];
+
+            $view->vars['value'] = [];
+
+            foreach ($entities as $entity) {
+                // Get text value
+                $text = is_null($textProperty)
+                    ? $value = (string) $entity
+                    : $accessor->getValue($entity, $textProperty);
+
+                // Get primary key
+                $primaryKey = $accessor->getValue($entity, $options['primary_key']);
+                if (is_null($primaryKey)) {
```

Comment:
Even if the primary key is not null, the prefix must be added if the entity is not persisted.

If not, it will be considered as an existing entity on the next submit, and will not be found.

This should be:
`if (!$this->em->contains($entity)) {`